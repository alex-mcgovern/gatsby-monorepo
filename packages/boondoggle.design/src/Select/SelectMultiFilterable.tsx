import React from "react";
import type { DropdownItem } from "../..";
import { Box } from "../Box";
import type { InputProps } from "../Input";
import { Label } from "../Label";
import type { GetSprinklesArgs } from "../__css__/getSprinkles.css";
import { DropdownMenu } from "./components/DropdownMenu";
import { SelectInput } from "./components/SelectInput";
import { DownshiftProviderMulti } from "./context/DownshiftProviderMulti";

export interface SelectMultiFilterableProps
  extends Omit<InputProps, "value" | "iconTrailing">,
    GetSprinklesArgs {
  items: Array<DropdownItem>;
  initialValue?: Array<DropdownItem>;
  onValueChange?(...args: Array<unknown>): unknown;
  onSelectItem?(...args: Array<unknown>): unknown;
  selectedItems?: Array<DropdownItem>;
  label?: string;
}

export function SelectMultiFilterable({
  items,
  onValueChange,
  onSelectItem,
  initialValue,
  id,
  disabled,
  iconLeading,
  label,
  placeholder,
  size,
  name,
  ...rest
}: SelectMultiFilterableProps) {
  return (
    <DownshiftProviderMulti
      items={items}
      onValueChange={onValueChange}
      onSelectItem={onSelectItem}
      initialValue={initialValue}
    >
      <Box position="relative" width="100%" {...rest}>
        {label && <Label htmlFor={id} label={label} />}
        <SelectInput
          disabled={disabled}
          iconLeading={iconLeading}
          id={id}
          name={name}
          placeholder={placeholder}
          size={size}
        />
        <DropdownMenu size={size} isSelectMultiFilterable />
      </Box>
    </DownshiftProviderMulti>
  );
}
