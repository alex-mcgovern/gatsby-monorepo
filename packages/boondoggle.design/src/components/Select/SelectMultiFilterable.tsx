import React from "react";
import type { GetSprinklesArgs } from "../../styles/getSprinkles.css";
import { Box } from "../Box";
import type { InputProps } from "../Input";
import { Label } from "../Label";
import { DropdownMenu } from "./components/DropdownMenu";
import { SelectInput } from "./components/SelectInput";
import { DownshiftProviderMulti } from "./context/DownshiftProviderMulti";
import type { DropdownItem } from "./types";

export interface SelectMultiFilterableProps
  extends Omit<InputProps, "value" | "iconTrailing">,
    GetSprinklesArgs {
  items: Array<DropdownItem>;
  initialValue?: Array<DropdownItem>;
  onValueChange?(...args: Array<unknown>): unknown;
  onSelectItem?(...args: Array<unknown>): unknown;
  selectedItems?: Array<DropdownItem>;
  label?: string;
}

export function SelectMultiFilterable({
  items,
  onValueChange,
  onSelectItem,
  initialValue,
  id,
  disabled,
  iconLeading,
  label,
  placeholder,
  size,
  name,
  ...rest
}: SelectMultiFilterableProps) {
  return (
    <DownshiftProviderMulti
      items={items}
      onValueChange={onValueChange}
      onSelectItem={onSelectItem}
      initialValue={initialValue}
    >
      <Box position="relative" width="100%" {...rest}>
        {label && <Label htmlFor={id} label={label} />}
        <SelectInput
          disabled={disabled}
          iconLeading={iconLeading}
          id={id}
          name={name}
          placeholder={placeholder}
          size={size}
        />
        <DropdownMenu size={size} isSelectMultiFilterable />
      </Box>
    </DownshiftProviderMulti>
  );
}
