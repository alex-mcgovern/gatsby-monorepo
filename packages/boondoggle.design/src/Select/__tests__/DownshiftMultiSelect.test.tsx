/** @jest-environment jsdom */
import React from "react";
import { cleanup } from "@testing-library/react";
import { renderTestComponent } from "~test/renderTestComponent";
import type { SelectMultiFilterableProps } from "../SelectMultiFilterable";
import { SelectMultiFilterable } from "../SelectMultiFilterable";

afterEach(cleanup);

test("Renders without error", async () => {
  const { getByTestId } = renderTestComponent<SelectMultiFilterableProps>(
    <SelectMultiFilterable
      id="test-dropdown"
      items={[
        { value: "foo", label: "foo" },
        { value: "bar", label: "bar" },
        { value: "bob", label: "bob" },
        { value: "alice", label: "alice" },
      ]}
      name="test-dropdown"
      size={undefined}
    />
  );

  expect(getByTestId("tested-component")).not.toBeNull();
  expect(getByTestId("tested-component")).toMatchSnapshot();
});
