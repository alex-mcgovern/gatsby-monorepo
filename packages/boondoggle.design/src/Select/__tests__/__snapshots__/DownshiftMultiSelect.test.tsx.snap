// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without error 1`] = `
<div
  aria-expanded="false"
  aria-haspopup="listbox"
  aria-owns="downshift-0-menu"
  class="get_sprinkles_position_relative__1reh6z97 get_sprinkles_width_100%_mobile__1reh6z107"
  data-testid="tested-component"
  role="combobox"
>
  <div
    class="get_sprinkles_position_relative__1reh6z97"
  >
    <input
      aria-autocomplete="list"
      aria-controls="downshift-0-menu"
      aria-labelledby="downshift-0-label"
      autocomplete="off"
      class="InputBase_getInputWrapperStyles__5mrrr91 InputBase__5mrrr90 get_sprinkles_paddingTop_spacing2_mobile__1reh6ztg get_sprinkles_paddingBottom_spacing2_mobile__1reh6znp get_sprinkles_paddingLeft_spacing2_mobile__1reh6zpm get_sprinkles_paddingRight_spacing2_mobile__1reh6zrj get_sprinkles_width_100%_mobile__1reh6z107 focus_ring_styles_focusRingStyles__xbo52k0 disabled_state_styles_disabledStateStyles__uj6nll0 InputBase_getInputWrapperStyles_size_md__5mrrr93 variant_interactive_element_size_variantInteractiveElementSize_md__1dzyhqw9 variant_interactive_element_size__1dzyhqw1 get_sprinkles_fontSize_body_lg_mobile__1reh6zfg get_sprinkles_fontSize_body_sm_desktop__1reh6zfo get_sprinkles_lineHeight_body_lg_mobile__1reh6zgs get_sprinkles_lineHeight_body_sm_desktop__1reh6zh0 get_sprinkles_height_spacing6_mobile__1reh6zbd get_sprinkles_height_spacing5_desktop__1reh6zbc"
      id="test-dropdown"
      name="test-dropdown"
      placeholder="All"
      value=""
    />
  </div>
  <div
    aria-labelledby="downshift-0-label"
    class="DownshiftMenu_dropdownListWrapper__stbbpv1 focus_ring_styles_focusRingStyles__xbo52k0 DownshiftMenu__stbbpv0 get_sprinkles_marginTop_spacing1_mobile__1reh6zlm get_sprinkles_marginBottom_none_mobile__1reh6zcy get_sprinkles_marginLeft_none_mobile__1reh6zhm get_sprinkles_marginRight_none_mobile__1reh6zjj get_sprinkles_width_max-content_mobile__1reh6z10v get_sprinkles_minWidth_100%__1reh6z8d get_sprinkles_display_block_mobile__1reh6zcg get_sprinkles_borderRadius_sm__1reh6z1j get_sprinkles_overflow_hidden__1reh6z8x get_sprinkles_background_background_1__1reh6z16g get_sprinkles_boxShadowLight__1reh6z1r get_sprinkles_border_dividerColorLight__1reh6z17 get_sprinkles_overflowY_auto__1reh6z8z DownshiftMenu_dropdownWrapperClosed__stbbpv4"
    id="downshift-0-menu"
    role="listbox"
  >
    <span
      aria-selected="false"
      class="DownshiftListItem_getDropdownStyles__ia0dkx1 DownshiftListItem__ia0dkx0 get_sprinkles_marginTop_none_mobile__1reh6zlg get_sprinkles_marginBottom_none_mobile__1reh6zcy get_sprinkles_marginLeft_none_mobile__1reh6zhm get_sprinkles_marginRight_none_mobile__1reh6zjj get_sprinkles_paddingTop_spacing2_mobile__1reh6ztg get_sprinkles_paddingBottom_spacing2_mobile__1reh6znp get_sprinkles_paddingLeft_spacing2_mobile__1reh6zpm get_sprinkles_paddingRight_spacing2_mobile__1reh6zrj get_sprinkles_display_flex_mobile__1reh6zcd get_sprinkles_alignItems_center__1reh6zz get_sprinkles_textDecoration_none__1reh6z17l get_sprinkles_width_100%_mobile__1reh6z107 get_sprinkles_textAlign_left_mobile__1reh6z13s get_sprinkles_color_text_highContrast__1reh6zi get_sprinkles_whiteSpace_nowrap__1reh6z9d focus_ring_styles_focusRingStyles__xbo52k0 disabled_state_styles_disabledStateStyles__uj6nll0 DownshiftListItem_getDropdownStyles_size_md__ia0dkx3 variant_interactive_element_size_variantInteractiveElementSize_md__1dzyhqw9 variant_interactive_element_size__1dzyhqw1 get_sprinkles_fontSize_body_lg_mobile__1reh6zfg get_sprinkles_fontSize_body_sm_desktop__1reh6zfo get_sprinkles_lineHeight_body_lg_mobile__1reh6zgs get_sprinkles_lineHeight_body_sm_desktop__1reh6zh0 get_sprinkles_height_spacing6_mobile__1reh6zbd get_sprinkles_height_spacing5_desktop__1reh6zbc"
      id="downshift-0-item-0"
      role="option"
    >
      <div
        class="get_sprinkles_display_flex_mobile__1reh6zcd get_sprinkles_alignItems_center__1reh6zz get_sprinkles_gap_spacing1__1reh6z4h"
      >
        <input
          readonly=""
          tabindex="-1"
          type="checkbox"
        />
        foo
      </div>
    </span>
    <span
      aria-selected="false"
      class="DownshiftListItem_getDropdownStyles__ia0dkx1 DownshiftListItem__ia0dkx0 get_sprinkles_marginTop_none_mobile__1reh6zlg get_sprinkles_marginBottom_none_mobile__1reh6zcy get_sprinkles_marginLeft_none_mobile__1reh6zhm get_sprinkles_marginRight_none_mobile__1reh6zjj get_sprinkles_paddingTop_spacing2_mobile__1reh6ztg get_sprinkles_paddingBottom_spacing2_mobile__1reh6znp get_sprinkles_paddingLeft_spacing2_mobile__1reh6zpm get_sprinkles_paddingRight_spacing2_mobile__1reh6zrj get_sprinkles_display_flex_mobile__1reh6zcd get_sprinkles_alignItems_center__1reh6zz get_sprinkles_textDecoration_none__1reh6z17l get_sprinkles_width_100%_mobile__1reh6z107 get_sprinkles_textAlign_left_mobile__1reh6z13s get_sprinkles_color_text_highContrast__1reh6zi get_sprinkles_whiteSpace_nowrap__1reh6z9d focus_ring_styles_focusRingStyles__xbo52k0 disabled_state_styles_disabledStateStyles__uj6nll0 DownshiftListItem_getDropdownStyles_size_md__ia0dkx3 variant_interactive_element_size_variantInteractiveElementSize_md__1dzyhqw9 variant_interactive_element_size__1dzyhqw1 get_sprinkles_fontSize_body_lg_mobile__1reh6zfg get_sprinkles_fontSize_body_sm_desktop__1reh6zfo get_sprinkles_lineHeight_body_lg_mobile__1reh6zgs get_sprinkles_lineHeight_body_sm_desktop__1reh6zh0 get_sprinkles_height_spacing6_mobile__1reh6zbd get_sprinkles_height_spacing5_desktop__1reh6zbc"
      id="downshift-0-item-1"
      role="option"
    >
      <div
        class="get_sprinkles_display_flex_mobile__1reh6zcd get_sprinkles_alignItems_center__1reh6zz get_sprinkles_gap_spacing1__1reh6z4h"
      >
        <input
          readonly=""
          tabindex="-1"
          type="checkbox"
        />
        bar
      </div>
    </span>
    <span
      aria-selected="false"
      class="DownshiftListItem_getDropdownStyles__ia0dkx1 DownshiftListItem__ia0dkx0 get_sprinkles_marginTop_none_mobile__1reh6zlg get_sprinkles_marginBottom_none_mobile__1reh6zcy get_sprinkles_marginLeft_none_mobile__1reh6zhm get_sprinkles_marginRight_none_mobile__1reh6zjj get_sprinkles_paddingTop_spacing2_mobile__1reh6ztg get_sprinkles_paddingBottom_spacing2_mobile__1reh6znp get_sprinkles_paddingLeft_spacing2_mobile__1reh6zpm get_sprinkles_paddingRight_spacing2_mobile__1reh6zrj get_sprinkles_display_flex_mobile__1reh6zcd get_sprinkles_alignItems_center__1reh6zz get_sprinkles_textDecoration_none__1reh6z17l get_sprinkles_width_100%_mobile__1reh6z107 get_sprinkles_textAlign_left_mobile__1reh6z13s get_sprinkles_color_text_highContrast__1reh6zi get_sprinkles_whiteSpace_nowrap__1reh6z9d focus_ring_styles_focusRingStyles__xbo52k0 disabled_state_styles_disabledStateStyles__uj6nll0 DownshiftListItem_getDropdownStyles_size_md__ia0dkx3 variant_interactive_element_size_variantInteractiveElementSize_md__1dzyhqw9 variant_interactive_element_size__1dzyhqw1 get_sprinkles_fontSize_body_lg_mobile__1reh6zfg get_sprinkles_fontSize_body_sm_desktop__1reh6zfo get_sprinkles_lineHeight_body_lg_mobile__1reh6zgs get_sprinkles_lineHeight_body_sm_desktop__1reh6zh0 get_sprinkles_height_spacing6_mobile__1reh6zbd get_sprinkles_height_spacing5_desktop__1reh6zbc"
      id="downshift-0-item-2"
      role="option"
    >
      <div
        class="get_sprinkles_display_flex_mobile__1reh6zcd get_sprinkles_alignItems_center__1reh6zz get_sprinkles_gap_spacing1__1reh6z4h"
      >
        <input
          readonly=""
          tabindex="-1"
          type="checkbox"
        />
        bob
      </div>
    </span>
    <span
      aria-selected="false"
      class="DownshiftListItem_getDropdownStyles__ia0dkx1 DownshiftListItem__ia0dkx0 get_sprinkles_marginTop_none_mobile__1reh6zlg get_sprinkles_marginBottom_none_mobile__1reh6zcy get_sprinkles_marginLeft_none_mobile__1reh6zhm get_sprinkles_marginRight_none_mobile__1reh6zjj get_sprinkles_paddingTop_spacing2_mobile__1reh6ztg get_sprinkles_paddingBottom_spacing2_mobile__1reh6znp get_sprinkles_paddingLeft_spacing2_mobile__1reh6zpm get_sprinkles_paddingRight_spacing2_mobile__1reh6zrj get_sprinkles_display_flex_mobile__1reh6zcd get_sprinkles_alignItems_center__1reh6zz get_sprinkles_textDecoration_none__1reh6z17l get_sprinkles_width_100%_mobile__1reh6z107 get_sprinkles_textAlign_left_mobile__1reh6z13s get_sprinkles_color_text_highContrast__1reh6zi get_sprinkles_whiteSpace_nowrap__1reh6z9d focus_ring_styles_focusRingStyles__xbo52k0 disabled_state_styles_disabledStateStyles__uj6nll0 DownshiftListItem_getDropdownStyles_size_md__ia0dkx3 variant_interactive_element_size_variantInteractiveElementSize_md__1dzyhqw9 variant_interactive_element_size__1dzyhqw1 get_sprinkles_fontSize_body_lg_mobile__1reh6zfg get_sprinkles_fontSize_body_sm_desktop__1reh6zfo get_sprinkles_lineHeight_body_lg_mobile__1reh6zgs get_sprinkles_lineHeight_body_sm_desktop__1reh6zh0 get_sprinkles_height_spacing6_mobile__1reh6zbd get_sprinkles_height_spacing5_desktop__1reh6zbc"
      id="downshift-0-item-3"
      role="option"
    >
      <div
        class="get_sprinkles_display_flex_mobile__1reh6zcd get_sprinkles_alignItems_center__1reh6zz get_sprinkles_gap_spacing1__1reh6z4h"
      >
        <input
          readonly=""
          tabindex="-1"
          type="checkbox"
        />
        alice
      </div>
    </span>
  </div>
</div>
`;
